bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ErrorValue.java

```diff
@@ -88,16 +88,31 @@ public ErrorValue(BType type, BString reason, Object details) {
     @Override
     @Deprecated
     public String stringValue() {
-        return "error " + reason.getValue() +
-                Optional.ofNullable(details).map(
-                        details -> " " + org.ballerinalang.jvm.values.utils.StringUtils.getStringValue(details)).orElse(
-                        "");
+        String detailsString;
+        if (details == null) {
+            detailsString = "";
+        } else {
+            detailsString = org.ballerinalang.jvm.values.utils.StringUtils.getStringValue(details);
+        }
+        if (detailsString.trim().isEmpty()) {
+            return "error " + reason.getValue();
```

Comment:
Do we need this conditional check? The `details` will be `null` if the user hasn't specified any details right?

Reply:
Yes, please check the new logic.

Reply:
We need to check because sometimes this is an empty MapValueImpl object